pkg/filter/proxy/proxy.go

```diff
@@ -315,6 +351,35 @@ func (b *Proxy) Handle(ctx context.HTTPContext) (result string) {
 	return ctx.CallNextHandler(result)
 }
 
+func (b *Proxy) updateAndGetClient() *Client {
+	super := b.filterSpec.Super()
+	if super == nil {
+		return b.client
+	}
+	tracing := super.Tracing()
+	// Read lock is enough in most of the cases as tracing is not initialized or updated often
+	b.clientMutex.RLock()
+	alreadyInit := !tracing.IsNoopTracer() && b.client.IsTracingInitialized()
+	noUpdate := tracing.IsNoopTracer() && !b.client.IsTracingInitialized()
+	b.clientMutex.RUnlock()
+
+	if alreadyInit || noUpdate {
+		return b.client
+	}
+	// Write lock to block readers while updating b.client
+	b.clientMutex.Lock()
+	defer b.clientMutex.Unlock()
+
+	if !tracing.IsNoopTracer() && !b.client.IsTracingInitialized() {
+		// tracing not ready yet
+		b.client = NewClient(b.client.client, tracing)
+	} else if tracing.IsNoopTracer() && b.client.IsTracingInitialized() {
+		// if HTTPServer is updated and there is no longer tracing configured
```

Comment:
`super.Tracing` is used as a singleton is EG, so if more than one HTTPServers are configured, and they are configured with different tracing options, what will the `super.Tracing` be?

Reply:
I changed proxy to use the tracing instance of a HTTPServer, so each HTTPServer can have different tracing options. Also HTTPServer passes tracing instance to proxy via Context and supervisor is not used for that.